test/e2e/util_test.go

```diff
@@ -545,20 +563,50 @@ func verifyInternalIngressController(t *testing.T, name types.NamespacedName, ho
 		"--retry-delay", "20",
 		"--max-time", "10",
 	}
-	clientPod := buildCurlPod("curl-"+name.Name, echoRoute.Namespace, image, address, echoRoute.Spec.Host, extraArgs...)
+	clientPodName := types.NamespacedName{echoRoute.Namespace, "curl-" + name.Name}
+	clientPodSpec := buildCurlPod(clientPodName.Name, clientPodName.Namespace, image, address, echoRoute.Spec.Host, extraArgs...)
+	clientPod := clientPodSpec.DeepCopy()
 	if err := kclient.Create(context.TODO(), clientPod); err != nil {
-		t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+		t.Fatalf("failed to create pod %q: %v", clientPodName, err)
 	}
 	defer func() {
 		if err := kclient.Delete(context.TODO(), clientPod); err != nil {
 			if errors.IsNotFound(err) {
 				return
 			}
-			t.Fatalf("failed to delete pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+			t.Fatalf("failed to delete pod %q: %v", clientPodName, err)
 		}
 	}()
 
-	err = wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+	curlPodLogs := ""
+	err = wait.PollImmediate(10*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), clientPodName, clientPod); err != nil {
+			t.Logf("error getting client pod %q: %v, retrying...", clientPodName, err)
+			return false, nil
+		}
+		// First check if client curl pod is still starting or not running.
+		if clientPod.Status.Phase == corev1.PodPending {
+			t.Logf("waiting for client pod %q to start", clientPodName)
+			return false, nil
+		} else if clientPod.Status.Phase == corev1.PodFailed {
+			// If not running, the pod most likely exited, but didn't provide a 200 response, let's try again.
+			t.Logf("client pod %q has failed...restarting", clientPodName)
+			if err := kclient.Delete(context.TODO(), clientPod); err != nil && errors.IsNotFound(err) {
+				t.Fatalf("failed to delete pod %q: %v", clientPodName, err)
+			}
+			// Wait for deletion to prevent a race condition.
+			err = kclient.Get(context.TODO(), clientPodName, clientPod)
+			for !errors.IsNotFound(err) {
+				t.Logf("waiting for %q: to be deleted", clientPodName)
+				err = kclient.Get(context.TODO(), clientPodName, clientPod)
+				time.Sleep(5 * time.Second)
```

Comment:
Another `pollImmediate` with a 5s timeout is always better than a 5s sleep.

Reply:
Andy said something similar, hopefully this helps address what you mean: https://github.com/openshift/cluster-ingress-operator/pull/845/files/bcf092c41b7cd5d1e4b74197fe2d45cbda038d6e#r999840724.

Reply:
That link doesn't work for me...

Reply:
Oh I probably linked on a specific changed, and updated that so it disappeared: 
https://github.com/openshift/cluster-ingress-operator/pull/845#discussion_r999732895